irohad/ametsuchi/impl/block_storage.hpp

```diff
@@ -79,74 +69,40 @@ namespace iroha {
        */
       bool add(Identifier id, const std::vector<uint8_t> &blob);
 
+      const std::string& directory() const;
+
       /**
        * Get data associated with
        * @param id - reference key
        * @return - blob, if exists
        */
-      nonstd::optional<std::vector<uint8_t>> get(Identifier id) const;
-
-      /**
-       * @return folder of storage
-       */
-      std::string directory() const;
+      boost::optional<std::vector<uint8_t>> get(Identifier id) const;
 
       /**
-       * @return maximal not null key
+       * @return number of identifiers in block storage.
        */
-      Identifier last_id() const;
+      size_t total_keys() const;
```

Comment:
Why not Identifier type?

Reply:
Identifier = type of key, when `total_keys` returns a number of keys  stored in the database.